provider-service/vai/internal/event_flow_unit_test.go

```diff
@@ -31,9 +32,21 @@ func artifact() *aiplatformpb.Artifact {
 	}
 }
 
+type MockPipelineJobClient struct {
+	mock.Mock
+}
+
+func (m *MockPipelineJobClient) GetPipelineJob(
+	ctx context.Context,
+	req *aiplatformpb.GetPipelineJobRequest,
+	opts ...gax.CallOption,
+) (*aiplatformpb.PipelineJob, error) {
+	args := m.Called(ctx, req, opts)
+	return args.Get(0).(*aiplatformpb.PipelineJob), args.Error(1)
```

Comment:
```suggestion
	return args.Get(0).(*aiplatformpb.PipelineJob), args.Error(1)
```

rather than always trying to cast to a `PipelineJob`, we should check if it's `nil` and return nil if so, this way we  don't have to pass an empty struct through just to satisfy the casting

Reply:
https://github.com/sky-uk/kfp-operator/pull/425/commits/0d8d38d8d638a3a5e5cc3ec8741b207d8112b763
https://github.com/sky-uk/kfp-operator/pull/425/commits/f338ac4486665ec039a129c69bc4162c3b1d0d8b